### PR TITLE
[tune] Remove rllib dep again, and add a test

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Load rllib as late as possible and only if needed (for RLlib agents). Add a test to make sure this doesn't regress again.

cc @vlad17

### CODE REVIEW

**Comment by richardliaw** [comment]
what was the current issue? don't the failed imports get caught in `register_all`?

**Comment by ericl** [comment]
Oh, it looks like the try catch is gone there. Hm, we could put it back, but this is probably a cleaner fix.

**Review by richardliaw — APPROVED** [review]

File: python/ray/tune/trial.py [hunk 1]
```diff
@@ -83,9 +88,11 @@ def __init__(
         in ray.tune.config_parser.
         """
 
-        if not ray.tune.registry._default_registry.contains(
-                ray.tune.registry.TRAINABLE_CLASS, trainable_name):
-            raise TuneError("Unknown trainable: " + trainable_name)
+        if not has_trainable(trainable_name):
+            # Make sure rllib agents are registered
+            from ray import rllib  # noqa # pylint: disable=unused-import
```
Comment (robertnishihara) [inline python/ray/tune/trial.py]: Nite, the comment should just be `# noqa: F401`

File: python/ray/tune/test/dependency_test.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
```
Comment (robertnishihara) [inline python/ray/tune/test/dependency_test.py]: This is fine if this is the only thing that's ever going to be in this test file. However in general new tests should all use `pytest`.

File: python/ray/tune/trial.py (hunk 1 above)
Comment (ericl) [reply python/ray/tune/trial.py]: Changed

File: python/ray/tune/test/dependency_test.py (hunk 2 above)
Comment (ericl) [reply python/ray/tune/test/dependency_test.py]: Yeah this one kinda has to be its own file since it tests import state.

**Pull request merged**